lib/puppet/type/schedule.rb

```diff
@@ -369,8 +369,8 @@ def match?(previous, now)
       validate do |values|
         values = [values] unless values.is_a?(Array)
         values.each { |value|
-          unless value.is_a?(String) and
-              (value =~ /^[0-6]$/ or value =~ /^(Mon|Tues?|Wed(?:nes)?|Thu(?:rs)?|Fri|Sat(?:ur)?|Sun)(day)?$/i)
+          unless (value.is_a?(Integer) and (0..6).include?(value)) or (value.is_a?(String) and
```

Comment:
This is really complex now and would be easier to read if you extracted a couple of functions to test valid strings or integers.

Reply:
+1

Reply:
I agree, it was hard for my brain just writing it.